### PR TITLE
Add locking to atomic operations in rw/rcu tests

### PR DESCRIPTION
**Pull request by nhorman**
I neglected to add locks to the calls to CRYPTO_atomic_add in these test, which on newer compilers is fine, as atomic operations are defined.  However on older compilers the __ATOMIC_ACQ_REL definition is missing causing these function to be implemented using an rwlock, which when NULL causes the locks to fail.

Fix this my creating the lock and using them appropriately

### CODE REVIEW

**Comment by mattcaswell** [comment]
Can we add "Fixes #24000" in the commit line?

File: crypto/sleep.c [hunk 1]
```diff
@@ -22,11 +22,20 @@ void OSSL_sleep(uint64_t millis)
...
     /* HPNS does not support usleep for non threaded apps */
     PROCESS_DELAY_(millis * 1000);
+#  elif defined(_SPT_MODEL_)
+#   include <spthread.h>
+#   include <spt_extensions.h>
+
+    usleep(millis * 1000);
+#  else
+    usleep(millis * 1000);
+#  endif
```
Comment (mattcaswell) [inline crypto/sleep.c:38]: This change looks unexpected? Is this meant to be here?

**Review by mattcaswell — CHANGES_REQUESTED** [review]

File: crypto/sleep.c (hunk 1 above)
Comment (nhorman) [reply crypto/sleep.c:38]: crud, no.  This is what happens when I do git add --all without doing a git status after :).  Will fix

**Review by paulidale — APPROVED** [review]

**Review by mattcaswell — APPROVED** [review]

**Comment by nhorman** [comment]
merged

**Comment by t8m** [comment]
Cherry picked to 3.3 branch as it was approved for it.

**Pull request closed without merging**